File: 프로그래머스/1/142086. 가장 가까운 같은 글자/가장 가까운 같은 글자.java
import java.util.*;
class Solution {
    public int[] solution(String s) {
        int[] answer = new int[s.length()];
        
        Map<String, Integer> map = new HashMap<>();
        
        for(int i = 0; i < s.length(); i++){
            char c = s.charAt(i);
            if(map.containsKey("" + c)){
                int index = i - map.get("" + c);
                map.put("" + c, i);
                answer[i] = index;
            }else {
                map.put("" + c, i);
                answer[i] = -1;
            }
        }
        
        return answer;
    }
}

File: 프로그래머스/1/142086. 가장 가까운 같은 글자/가장 가까운 같은 글자.py
def solution(s):
    answer = []
    check = [-1] * 26
    
    for i, c in enumerate(s):
        index = ord(c) - ord('a')
        
        if check[index] == -1:
            check[index] = i;
            answer.append(-1)
        else:
            answer.append(i - check[index])
        check[index] = i;
        
    # 1)) 평균 실행 0.58ms, 메모리 사용량 9.38mb
            
    return answer